### PR TITLE
introduce stand-in indexer

### PR DESCRIPTION
**Pull request by mtbc**
With Hibernate 3.6 the OMERO 5.4 indexer fails with a Sealed WorkQueue exception. This PR introduces a stand-in replacement to index adequately until the problem can be better addressed. See https://trello.com/c/XdwEzrvs/73-indexer-log-hibernate-error.

CLI foreground reindexing (with `--foreground`) is *not* fixed but background should suffice.

### CODE REVIEW

**Comment by mtbc** [comment]
To see more of its activity can add to `etc/logback-indexing.xml`,
```xml
<logger name="ome.services.fulltext.FullTextIndexer2" level="DEBUG"/>
```

**Comment by joshmoore** [comment]
Currently testing via https://github.com/ome/omero-build/pull/6

**Comment by mtbc** [comment]
Now running into just a couple of integration test failures, mostly intermittent. Probably nothing major, will investigate. Consistent failure is of `testFilename`.

File: src/main/java/ome/services/fulltext/FullTextIndexer2.java [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+import org.quartz.Scheduler;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.quartz.MethodInvokingJobDetailFactoryBean;
+
+/**
+ * An indexer bean replacing the 5.4 full-text thread with adequate functionality.
+ * Exists as a stand-in while Hibernate / Spring upgrade issues remain unresolved.
```
Comment (joshmoore) [inline src/main/java/ome/services/fulltext/FullTextIndexer2.java]: @mtbc : could you extend this a bit with:
 - how this differs from the previous indexer (trade-offs, etc)
 - and how the implementation functions ("first A, then ...")

**Comment by mtbc** [comment]
The issue with `test3164*` seems somehow related to the use of `IUpdate.indexObject`.

**Comment by mtbc** [comment]
Ha, `IUpdate.indexObject` fires up the legacy indexer. For now I'll tweak it to instead create a REINDEX entry in the event log. Alternatively could plug actual indexer more clearly into update service if preferred.

**Comment by mtbc** [comment]
Heh, so: interesting issue: my simple approach to fixing `IUpdate.indexObject` fails because some tests like `test3721Ordering` expect that method to index things like tags which are *not* in `omero.search.include_types` so the indexer ignores those requests. (They would get indexed if the request were to index a tagged image but that test doesn't have such.) I assume that `IUpdate.indexObject` should override `include_types`? In which case, I suggest: expect some `IUpdate.indexObject`-using tests to fail until I push a fix tomorrow.

File: src/main/java/ome/services/fulltext/FullTextIndexer2.java (hunk 1 above)
Comment (mtbc) [reply src/main/java/ome/services/fulltext/FullTextIndexer2.java]: Sure, though I may miss some differences and maybe not in time for m4.

**Comment by joshmoore** [comment]
Green with jars from artifactory after conflict resolution. Merging proactively for M4 and the more significant testing can happen there.

File: src/main/java/ome/services/fulltext/FullTextIndexer2.java (hunk 1 above)
Comment (mtbc) [reply src/main/java/ome/services/fulltext/FullTextIndexer2.java]: Extended in #23.

**Pull request merged**